osmoutils/store_helper_test.go

```diff
@@ -186,3 +189,220 @@ func (s *TestSuite) TestNoStopFn_AlwaysFalse() {
+	}{
+		"basic valid Dec test": {
+			setKey: keyA,
+			setValue: &sdk.DecProto{
+				Dec: sdk.OneDec(),
+			},
+
+			actualResultProto: &sdk.DecProto{},
+		},
+		"basic valid TwapRecord test": {
+			setKey: keyA,
+			setValue: &twaptypes.TwapRecord{
+				PoolId: 2,
+			},
+
+			actualResultProto: &twaptypes.TwapRecord{},
+		},
+		"invalid set value": {
+			setKey:   keyA,
+			setValue: (*sdk.DecProto)(nil),
+
+			expectPanic: true,
+		},
+	}
+
+	for name, tc := range tests {
+		s.Run(name, func() {
+			s.SetupStoreWithBasePrefix()
+
+			// Setup
+			osmoassert.ConditionalPanic(s.T(), tc.expectPanic, func() {
+				osmoutils.MustSet(s.store, []byte(tc.setKey), tc.setValue)
+			})
+
+			if tc.expectPanic {
+				return
+			}
+
+			osmoutils.MustGet(s.store, []byte(tc.setKey), tc.actualResultProto)
+			s.Require().Equal(tc.setValue.String(), tc.actualResultProto.String())
+		})
+	}
+}
+
+// TestMustGetDec tests that MustGetDec retrieves the correct
+// decimal values from the store and panics if an error is encountered.
+func (s *TestSuite) TestMustGetDec() {
+
+	tests := map[string]struct {
+		// keyys and values to preset
+		preSetKeyValues map[string]sdk.Dec
+
+		// keys and values to attempt to get and validate
+		getKeyValues map[string]sdk.Dec
+
+		expectPanic bool
+	}{
+		"valid get": {
+			preSetKeyValues: map[string]sdk.Dec{
+				keyA: sdk.OneDec(),
+				keyB: sdk.OneDec().Add(sdk.OneDec()),
+				keyC: sdk.OneDec().Add(sdk.OneDec()).Add(sdk.OneDec()),
+			},
+
+			getKeyValues: map[string]sdk.Dec{
+				keyA: sdk.OneDec(),
+				keyB: sdk.OneDec().Add(sdk.OneDec()),
+				keyC: sdk.OneDec().Add(sdk.OneDec()).Add(sdk.OneDec()),
+			},
+		},
+		"attempt to get non-existent key - panic": {
+			preSetKeyValues: map[string]sdk.Dec{
+				keyA: sdk.OneDec(),
+				keyC: sdk.OneDec().Add(sdk.OneDec()).Add(sdk.OneDec()),
+			},
+
+			getKeyValues: map[string]sdk.Dec{
+				keyB: sdk.OneDec().Add(sdk.OneDec()),
+			},
+
+			expectPanic: true,
+		},
+	}
+
+	for name, tc := range tests {
+		s.Run(name, func() {
+			s.SetupStoreWithBasePrefix()
+
+			// Setup
+			for key, value := range tc.preSetKeyValues {
+				osmoutils.MustSetDec(s.store, []byte(key), value)
+			}
+
+			osmoassert.ConditionalPanic(s.T(), tc.expectPanic, func() {
+				for key, expectedValue := range tc.getKeyValues {
+					// System under test.
+					actualDec := osmoutils.MustGetDec(s.store, []byte(key))
+					// Assertions.
+					s.Require().Equal(expectedValue.String(), actualDec.String())
+				}
+			})
+		})
+	}
+}
+
+// TestMustSetDec tests that MustSetDec updates the store correctly
+// with the right decimal value.
+func (s *TestSuite) TestMustSetDec() {
+	// Setup.
+	s.SetupStoreWithBasePrefix()
+
+	originalDecValue := sdk.OneDec()
+
+	// System under test.
+	osmoutils.MustSetDec(s.store, []byte(keyA), originalDecValue)
+
+	// Assertions.
+	retrievedDecVaue := osmoutils.MustGetDec(s.store, []byte(keyA))
+	s.Require().Equal(originalDecValue.String(), retrievedDecVaue.String())
+}
```

Comment:
I assume we are not testing the fail case because MustSet is already covered right?

Reply:
No, I don't think it is possible to get a panic in this case because the only reason `MustSet` could panic if we provide an invalid proto type:
https://github.com/osmosis-labs/osmosis/blob/3e334d042f298112a5704be410b77e9f1432d0d9/osmoutils/store_helper.go#L81-L84

However, in `MustSetDec` we pretty much hardcoded a valid proto:
https://github.com/osmosis-labs/osmosis/blob/3e334d042f298112a5704be410b77e9f1432d0d9/osmoutils/store_helper.go#L122-L124

I will add a comment about this

Reply:
Ahh that makes sense, looks good then!